modules/core/src/main/scala/org/scalasteward/core/nurture/PullRequestRepository.scala

```diff
@@ -55,6 +66,28 @@ final class PullRequestRepository[F[_]](
       }
       .void
 
+  def getObsoleteOpenPullRequests(
+      repo: Repo,
+      updateData: Update
+  ): F[List[(Uri, Update)]] = {
+    val crossDependency = updateData match {
+      case Update.Single(crossDependency, _, _) =>
+        crossDependency
+      case Update.Group(crossDependencies, _) =>
+        crossDependencies.head
+    }
+
+    kvStore.get(repo).map {
+      _.getOrElse(Map.empty).collect {
+        case (url, data)
+            if UpdateAlg.isForSameArtifacts(data.update, crossDependency) &&
```

Comment:
I'm wondering if it wouldn't be easier to add `Update.withNewerVersions(versions: Nel[String]): Update` and then to use that here instead of `UpdateAlg.isForSameArtifacts` to find older updates, i.e.  `data.update.withNewerVersions(updateData.newerVersions) === updateData`? For `Update.Group` it would also mean that only other PRs that update the exact same list of artifacts are found. Currently all older PRs that contain the first artifactId in `updateData` are found.

Reply:
I introduced the changes you proposed, it makes the code cleaner.

Meanwhile I had a question, I implemented like so:
```
  def withNewerVersions(versions: Nel[String]): Update = this match {
    case s @ Single(_, _, _) =>
      s.copy(newerVersions = versions)
    case g @ Group(_, _) =>
      g.copy(newerVersions = versions)
  }
```
That works in practice but we loose some type information and I find it disappointing, maybe you have a better solution. The problem is that if we have some `val update: Update.Single = ???` and we do `val newUpdate = update.withNewerVersions(updateData.newerVersions)`, well `newUpdate` will have the type `Update` and not `Update.Single`

Would you know how to implement `withNewerVersions` without the loss of the precise type?

Reply:
Thanks for the change! I try to test it again tomorrow with a few more repos and will merge if nothing serious pops up.

Regarding your question. You could leave the implementation in `Update` abstract and override it in `Single` and `Group` with more precise return types:
```scala
class Single { override def withNewerVersions(versions: Nel[String]): Single = ??? }
class Group { override def withNewerVersions(versions: Nel[String]): Group = ??? }
```